plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/IcebergQueryRunner.java

```diff
@@ -46,64 +51,70 @@ public static DistributedQueryRunner createIcebergQueryRunner()
     }
 
     public static DistributedQueryRunner createIcebergQueryRunner(
             Map<String, String> extraProperties,
             Map<String, String> connectorProperties,
-            Iterable<TpchTable<?>> tables)
+            Optional<File> metastoreDirectory)
             throws Exception
     {
         return createIcebergQueryRunner(
                 extraProperties,
                 connectorProperties,
-                tables,
-                Optional.empty());
+                metastoreDirectory,
+                IcebergQueryRunner::initializeSchema);
     }
 
     public static DistributedQueryRunner createIcebergQueryRunner(
             Map<String, String> extraProperties,
             Map<String, String> connectorProperties,
-            Iterable<TpchTable<?>> tables,
-            Optional<File> metastoreDirectory)
+            Optional<File> metastoreDirectory,
+            Consumer<QueryRunner> schemaInitializer)
             throws Exception
     {
         Session session = testSessionBuilder()
                 .setCatalog(ICEBERG_CATALOG)
-                .setSchema("tpch")
+                .setSchema(SCHEMA)
                 .build();
 
         DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session)
                 .setExtraProperties(extraProperties)
                 .build();
 
         queryRunner.installPlugin(new TpchPlugin());
-        queryRunner.createCatalog("tpch", "tpch");
-
-        Path dataDir = metastoreDirectory.map(File::toPath).orElseGet(() -> queryRunner.getCoordinator().getBaseDataDir().resolve("iceberg_data"));
+        queryRunner.createCatalog("tpch", SCHEMA);
 
         queryRunner.installPlugin(new IcebergPlugin());
         connectorProperties = new HashMap<>(ImmutableMap.copyOf(connectorProperties));
-        connectorProperties.putIfAbsent("iceberg.catalog.type", "TESTING_FILE_METASTORE");
-        connectorProperties.putIfAbsent("hive.metastore.catalog.dir", dataDir.toString());
+        if (!connectorProperties.containsKey("iceberg.catalog.type")) {
+            Path dataDir = metastoreDirectory.map(File::toPath).orElseGet(() -> queryRunner.getCoordinator().getBaseDataDir().resolve("iceberg_data"));
+            connectorProperties.putIfAbsent("iceberg.catalog.type", "TESTING_FILE_METASTORE");
+            connectorProperties.putIfAbsent("hive.metastore.catalog.dir", dataDir.toString());
+        }
         queryRunner.createCatalog(ICEBERG_CATALOG, "iceberg", connectorProperties);
-
-        queryRunner.execute("CREATE SCHEMA tpch");
-
-        copyTpchTables(queryRunner, "tpch", TINY_SCHEMA_NAME, session, tables);
-
+        schemaInitializer.accept(queryRunner);
         return queryRunner;
     }
 
+    public static void initializeSchema(QueryRunner queryRunner)
+    {
+        queryRunner.execute("CREATE SCHEMA " + SCHEMA);
+        copyTpchTables(queryRunner, "tpch", TINY_SCHEMA_NAME, queryRunner.getDefaultSession(), ImmutableList.of(NATION, REGION));
```

Comment:
why nation & region?

if no tables were asked for, no tables should be created.

Reply:
I uderstood https://github.com/trinodb/trino/pull/10894#discussion_r802741292 it's a most common use case